components/operator/internal/resources/auths/controller.go

```diff
@@ -59,10 +60,20 @@ func Reconcile(ctx Context, stack *v1beta1.Stack, auth *v1beta1.Auth, version st
 			return errors.Wrap(err, "resolving image")
 		}
 
+		serviceAccountName, err := settings.GetAWSRole(ctx, stack.Name)
+		if err != nil {
+			return errors.Wrap(err, "getting service account name")
+		}
+
+		migrateContainer, err := databases.MigrateDatabaseContainer(ctx, stack, image, database)
+		if err != nil {
+			return errors.Wrap(err, "creating migrate container")
+		}
+
 		if IsGreaterOrEqual(version, "v2.0.0-rc.5") && databases.GetSavedModuleVersion(database) != version {
 			if err := jobs.Handle(ctx, auth, "migrate",
-				databases.MigrateDatabaseContainer(image, database),
-				jobs.WithServiceAccount(database.Status.URI.Query().Get("awsRole")),
+				migrateContainer,
+				jobs.WithServiceAccount(serviceAccountName),
```

Comment:
retrieving serviceAccountName & migrateContainer is not needed when version < v2.0.0-rc.5"